server/db/db.js

```diff
@@ -89,6 +91,10 @@ async function beginTransaction(retryCount = 0) {
   return pgClient
 }
 
+async function beginTransaction() {
+  return runBeginTransactionWithRetries(0)
```

Comment:
I believe this should be `return await runBeginTransactionWithRetries(0)`. As `beginTransaction` is an `async` function, it returns a Promise that is resolved upon a return statement; but as `runBeginTransactionWithRetries` is also an async function, and it is being returned without awaiting, then `beginTransaction` returns a Promise that when awaited gives another Promise... which I don't think is what you want.

Reply:
I think it's possible that because there are no `await`s used in `beginTransaction`, it being `async` is being ignored, and the first level Promise is not occurring at all, and the Promise from `runBeginTransactionWithRetries` is being returned instead. However, I think it'd be good to be more explicit in the code on how the Promises are being handled, than to hope JavaScript interprets this in such a way! (I.e., add the `await` before `runBeginTransactionWithRetries`.)

Reply:
Thanks Micah, I actually originally had `return await runBeginTransactionWithRetries(0)` in my code, but adding `await` caused an ESLint error, saying that adding `await` added redundancy, which is why it was taken out. 

I looked into the ESLint error here: https://eslint.org/docs/latest/rules/no-return-await and it looks like this rule was eventually deprecated. What do you think about me changing the code to the following so that I can include `await` and bypass the ESLint error: 
```
async function beginTransaction() {
  const pgClient = await runBeginTransactionWithRetries(0)
  return pgClient
}
```

Reply:
Oo, I think that's a great workaround!! I was thinking an alternative is to just remove `async` from `beginTransaction`, but then `await`ing a non-async function might make ESLint complain, even though `beginTransaction` always returns a Promise. The example you've just written is super clear and I prefer it to removing async from the function :) .

Reply:
Thanks Micah, I actually originally had `return await runBeginTransactionWithRetries(0)` in my code, but adding `await` caused an ESLint error, saying that adding `await` added redundancy, which is why it was taken out. 

I looked into the ESLint error here: https://eslint.org/docs/latest/rules/no-return-await and it looks like this rule was eventually deprecated. What do you think about changing the code to the following so that I can include `await` and bypass the ESLint error: 
```
async function beginTransaction() {
  try {
    const pgClient = await runBeginTransactionWithRetries(0)
    return pgClient
  } catch (e) {
    helpers.logError(`Would add a more descriptive error: ${e}`)
    return null
  }
}
```

Reply:
Updated!

Reply:
Yay!

Reply:
I also like the addition of catching any errors.